File: dsa_algo/leetcode/arrays/lc_arrays/java/findMaxConsecutiveOnes.java
class findMaxConsecutiveOnes {
    public static int f(int[] nums)
    {
        int max_count = 0;
        int current_count = 0;
        for(int i = 0; i < nums.length; i++)
        {
            if(nums[i] == 0)
            {
                max_count = Math.max(max_count, current_count);
                current_count = 0;
            }
            else
                current_count++;
        }
        return Math.max(max_count, current_count);
    }
    public static void main(String[] args)
    {
        int[] arr = {1, 1, 0, 1, 1, 1};
        System.out.println(findMaxConsecutiveOnes.f(arr));
    }
}

File: dsa_algo/leetcode/arrays/lc_arrays/c++/find_max_consecutive_ones.cpp
#include<bits/stdc++.h>
int max_consecutive_ones(std::vector<int>& v);

int main()
{
    std::vector<int> v = {1, 0, 1, 1, 0, 1};
    std::cout<<max_consecutive_ones(v);
    return 0;
}
int max_consecutive_ones(std::vector<int>& v)
{
    int max_count = 0;
    int current_count = 0;
    for(auto val: v)
    {
        if(val == 0)
        {
            max_count = std::max(max_count, current_count);
            current_count = 0;
        }
        else
            current_count++;
    }
    return std::max(max_count, current_count);
}
